src/shared/istiopolicy/creator.go

```diff
@@ -194,8 +195,35 @@ func (c *Creator) getPolicyName(intents *v1alpha2.ClientIntents, intent v1alpha2
 func (c *Creator) isPolicyEqual(existingPolicy *v1beta1.AuthorizationPolicy, newPolicy *v1beta1.AuthorizationPolicy) bool {
 	sameServer := existingPolicy.Spec.Selector.MatchLabels[v1alpha2.OtterizeServerLabelKey] == newPolicy.Spec.Selector.MatchLabels[v1alpha2.OtterizeServerLabelKey]
 	samePrincipals := existingPolicy.Spec.Rules[0].From[0].Source.Principals[0] == newPolicy.Spec.Rules[0].From[0].Source.Principals[0]
+	sameHTTPRules := compareHTTPRules(existingPolicy.Spec.Rules[0].To, newPolicy.Spec.Rules[0].To)
 
-	return sameServer && samePrincipals
+	return sameServer && samePrincipals && sameHTTPRules
+}
+
+func compareHTTPRules(existingRules []*v1beta12.Rule_To, newRules []*v1beta12.Rule_To) bool {
+	if len(existingRules) == 0 && len(newRules) == 0 {
+		return true
+	}
+
+	if len(existingRules) != len(newRules) {
+		return false
+	}
+
+	for i := range existingRules {
```

Comment:
If you're relying on the order for comparison, you should probably sort them, otherwise you risk an equal set of rules not actually being equal.
Or, place them in a map for comparison.